src/rrd_cgi.c

```diff
@@ -1379,8 +1406,10 @@ s_var   **rrdcgiReadVariables(
             length = atoi(ip);
             if ((line = (char *) malloc(length + 2)) == NULL)
                 return NULL;
-            if (fgets(line, length + 1, stdin) == NULL)
+            if (fgets(line, length + 1, stdin) == NULL) {
+            	free(line);
```

Comment:
Yeah my editor removes all whitespace so I had to hand-edit some commits not to add a lot of whitespace diff. There should be 4 spaces there.